src/webgpu/api/operation/texture_view/format_reinterpretation.spec.ts

```diff
@@ -0,0 +1,390 @@
+                                        vec2<f32>(-1.0, -1.0),
+                                        vec2<f32>(-1.0,  1.0),
+                                        vec2<f32>( 1.0, -1.0),
+                                        vec2<f32>(-1.0,  1.0),
+                                        vec2<f32>( 1.0, -1.0),
+                                        vec2<f32>( 1.0,  1.0));
+            return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+          }`,
+      }),
+      entryPoint: 'main',
+    },
+    fragment: {
+      module:
+        multisample.sample > 1
+          ? device.createShaderModule({
+              code: `
+            @group(0) @binding(0) var src: texture_multisampled_2d<f32>;
+            @stage(fragment) fn main(@builtin(position) coord: vec4<f32>) -> @location(0) vec4<f32> {
+              var result : vec4<f32>;
+              for (var i = 0; i < ${multisample.sample}; i = i + 1) {
+                result = result + textureLoad(src, vec2<i32>(coord.xy), i);
+              }
+              return result * ${1 / multisample.sample};
+            }`,
+            })
+          : device.createShaderModule({
+              code: `
+            @group(0) @binding(0) var src: texture_2d<f32>;
+            @stage(fragment) fn main(@builtin(position) coord: vec4<f32>) -> @location(0) vec4<f32> {
+              return textureLoad(src, vec2<i32>(coord.xy), 0);
+            }`,
+            }),
+      entryPoint: 'main',
+      targets: [{ format }],
+    },
+    multisample: {
+      count: multisample.render,
+    },
+  });
+}
+
+g.test('texture_binding')
+  .desc(`Test that a regular texture allocated as 'format' is correctly sampled as 'viewFormat'.`)
+  .params(u =>
+    u //
+      .combine('format', kRegularTextureFormats)
+      .combine('viewFormat', kRegularTextureFormats)
+      .filter(
+        ({ format, viewFormat }) => format !== viewFormat && viewCompatible(format, viewFormat)
+      )
+  )
+  .fn(async t => {
+    const { format, viewFormat } = t.params;
+
+    // Make an input texel view.
+    const inputTexelView = makeInputTexelView(format);
+
+    // Write data out to bytes.
+    const texelSize = kTextureFormatInfo[format].bytesPerBlock;
+    const textureData = new Uint8ClampedArray(texelSize * kTextureSize * kTextureSize);
+    inputTexelView.writeTextureData(textureData, {
+      bytesPerRow: texelSize * kTextureSize,
+      rowsPerImage: kTextureSize,
+      subrectOrigin: [0, 0],
+      subrectSize: [kTextureSize, kTextureSize],
+    });
+
+    // Create and upload data to the texture.
+    const texture = t.device.createTexture({
+      format,
+      size: [kTextureSize, kTextureSize],
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.TEXTURE_BINDING,
+      viewFormats: [viewFormat],
+    });
+
+    t.device.queue.writeTexture(
+      { texture },
+      textureData,
+      {
+        bytesPerRow: texelSize * kTextureSize,
+      },
+      [kTextureSize, kTextureSize]
+    );
```

Comment:
maybe factor out to a `makeTextureWithContents` like `makeBufferWithContents`. (Can just put it in gpu_test; I don't think we actually need the separate `makeBufferWithContents` function that we have now)

Generating data straight into a mappedAtCreation buffer and copying to the texture might also be a little more optimal, though it's an extra copy vs an extra (non-staging) buffer allocation so idk.